pkg/virt-launcher/virtwrap/cli/libvirt.go

```diff
@@ -525,50 +525,63 @@ func (l *LibvirtConnection) reconnectIfNecessary() (err error) {
 	defer l.reconnectLock.Unlock()
 	// TODO add a reconnect backoff, and immediately return an error in these cases
 	// We need this to avoid swamping libvirt with reconnect tries
-	if !l.alive {
-		l.Connect, err = newConnection(l.uri, l.user, l.pass)
-		if err != nil {
-			return
-		}
-		l.alive = true
+	if l.alive {
+		return nil
+	}
 
-		log.Log.Info("Established new Libvirt Connection")
+	if l.Connect, err = newConnection(l.uri, l.user, l.pass); err != nil {
+		return
```

Comment:
Can you avoid the naked return? (although it works the same)

Reply:
Yes, sure, done for all of them